android/src/main/java/io/invertase/firebase/notifications/RNFirebaseBackgroundNotificationActionReceiver.java

```diff
@@ -0,0 +1,50 @@
+package io.invertase.firebase.notifications;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.facebook.react.HeadlessJsTaskService;
+import com.facebook.react.ReactApplication;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.WritableMap;
+
+import io.invertase.firebase.Utils;
+
+public class RNFirebaseBackgroundNotificationActionReceiver extends BroadcastReceiver {
+  static boolean isBackgroundNotficationIntent(Intent intent) {
+    return intent.getExtras() != null && intent.hasExtra("action") && intent.hasExtra("notification");
+  }
+
+  static WritableMap toNotificationOpenMap(Intent intent) {
+    Bundle extras = intent.getExtras();
+    WritableMap notificationMap = Arguments.makeNativeMap(extras.getBundle("notification"));
+    WritableMap notificationOpenMap = Arguments.createMap();
+    notificationOpenMap.putString("action", extras.getString("action"));
+    notificationOpenMap.putMap("notification", notificationMap);
+    return notificationOpenMap;
+  }
+
+  @Override
+  public void onReceive(Context context, Intent intent) {
+    if (!isBackgroundNotficationIntent(intent)) {
+      return;
+    }
+
+    if (Utils.isAppInForeground(context)) {
```

Comment:
Have you been able to test that this works in the foreground?

Generally the broadcast receiver runs outside the react context, so I'd be surprised if this did work as it is currently.  The way we work around this in other areas is to use a local broadcast receiver which the react application listens to directly.  I'm more than happy to merge and then make these changes myself, but wanted to understand if you'd managed to get it working this way before I do?

Reply:
I've just retested it with [HabitChallenge](http://HabitChallenge.co/) app in three scenarios:
1. app in foreground
1. app in background
1. app not running

In all of them everything worked as expected.

Reply:
BTW. Test were done with `react-native` 0.55.4 and android simulator running Android 8.1.0

Reply:
Great! In which case, I will get this merged and we can keep an eye on it.
If we get reports of issues in the foreground then I can easily switch it over to use the local broadcast.

Thank you again for such a comprehensive PR!  I'll look to get some documentation added and we'll include this as part of a release today or tomorrow.

Reply:
Cool! Looking forward for the release! :D